infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -28,26 +28,29 @@ const (
 	FailedPhase  = "Failed"
 
 	// Feast condition types:
-	ClientReadyType       = "Client"
-	OfflineStoreReadyType = "OfflineStore"
-	OnlineStoreReadyType  = "OnlineStore"
-	RegistryReadyType     = "Registry"
-	ReadyType             = "FeatureStore"
+	ClientReadyType         = "Client"
+	OfflineStoreReadyType   = "OfflineStore"
+	OnlineStoreReadyType    = "OnlineStore"
+	RegistryReadyType       = "Registry"
+	ReadyType               = "FeatureStore"
+	KubernetesAuthReadyType = "KubernetesAuth"
```

Comment:
with oidc auth support, will the operator deploy anything? would a separate auth condition be set or would the k8s auth condition just be removed? 

if oidc will deploy anything, we should make this condition type more generic so both can use it. maybe `Authentication`. if this is only used for k8s auth, i think we're good as is.

Reply:
OIDC will not deploy anything but has to validate that the linked secret includes all the required parameters as secret keys. I was thinking of using the condition to mark the completion state of this step, do you think it's only meant to track the actual resource deployments instead?

Reply:
@dmartinol no, i think your line of thinking is a fair approach. in that case though, we should probably make the ReadyType more generic.

Reply:
What about a single `ReadyType` and two distinct `AuthFailedReason` and `AuthReadyMessage`?